ncat/ncat_core.c

```diff
@@ -171,6 +171,8 @@ void options_init(void)
     o.proto = IPPROTO_TCP;
     o.broker = 0;
     o.listen = 0;
+    o.delimiter = 0;
+    o.delimiter_used = 0; /* This gets activated when delimiter option gets enabled. (Works as a flag) */
```

Comment:
Since you're using an `int` for `delimiter` but only using one byte of it (casting to `char`), you can use the sign to represent this flag instead. If `o.delimiter` is less than 0, it is not set. Using a whole `int` is a little wasteful; you could use a `short` instead, since you're only using the sign and one byte.